### PR TITLE
[Analyzer] Do not force 64-bit architectures on Xcode 10 or higher

### LINKED ISSUE #8242: Xcode 10.1 warnings after install
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install`

## What did you expect to happen?

Install all pod dependencies correctly.

## What happened instead?

Warnings - Target XXXX overrides the Architecture setting

screen shot 2018-11-01 at 1 24 01 pm

### Stack

```
   CocoaPods : 1.6.0.beta.2
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 2.5.2.3
        Host : Mac OS X 10.14.1 (18B75)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : bitbucket-xxx-pods - https://bitbucket.org/xxx/pods.git @ f1590365fdcef1c933712de42509adb7cc81c695
               master - https://github.com/CocoaPods/Specs.git @ 2169acc71e90de4dbd70cb1121f2321b231910b5
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://bitbucket.org/XXXXXX/pods.git'
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '11.0'


def available_pods
    pod 'ALForm'
    pod 'Fabric'
    pod 'Crashlytics'
    pod 'extensions'
    pod 'Toast-Swift'
    pod 'SwiftValidators'
end

target 'LabelCheck' do
    use_frameworks!
    available_pods
    
    target 'LabelCheckTests' do
    	inherit! :complete
	end
end
```

## Project that demonstrates the issue

N/A

### PR DESCRIPTION
**Pull request by amorde**
Closes #8242

### CODE REVIEW

**Comment by amorde** [comment]
Looks like integration specs are affected. I'll see if I can have a couple of them use different object versions

**Comment by amorde** [comment]
Integration specs PR: https://github.com/CocoaPods/cocoapods-integration-specs/pull/194

**Comment by amorde** [comment]
@dnkoutso  @segiddins were the integration specs out of date? bumping to the latest SHA shows a lot more changes than what was in my PR

File: lib/cocoapods/installer/analyzer.rb [hunk 1]
```diff
@@ -22,6 +22,11 @@ class Analyzer
       #
       IOS_64_BIT_ONLY_VERSION = Version.new('11.0')
 
+      # @return [Integer] The Xcode object version until which 64-bit architectures should be manually specified
+      #
+      # Xcode 10 will automatically select the correct architectures based on deployment target
+      IOS_64_BIT_ONLY_PROJECT_VERSION = 50
```
Comment (amorde) [inline lib/cocoapods/installer/analyzer.rb]: is 50 correct here? or did Xcode 10 start with 51?

Comment (dnkoutso) [reply lib/cocoapods/installer/analyzer.rb]: yes see `target_integrator.rb`

```ruby
          def input_output_paths_use_filelist?(object)
            object.project.object_version.to_i >= 50
          end
```

Can we use this constant there too?

**Comment by dnkoutso** [comment]
Looks like integration specs havent been updated properly yes, PR related to multiple pods project

File: lib/cocoapods/installer/analyzer.rb [hunk 2]
```diff
@@ -702,7 +708,8 @@ def dependencies_for_specs(specs, platform, all_specs)
       # @return [PodTarget]
       #
       def generate_pod_target(target_definitions, target_inspections, specs, scope_suffix: nil, build_type: nil)
-        target_requires_64_bit = target_definitions.all? { |td| requires_64_bit_archs?(td.platform) }
+        object_version = target_inspections.values.map { |ti| ti.project.object_version }.min
```
Comment (dnkoutso) [inline lib/cocoapods/installer/analyzer.rb]: I guess a project can have multiple object versions? Maybe due to using different `.xcodeproj` I guess?

Comment (dnkoutso) [reply lib/cocoapods/installer/analyzer.rb]: if so is this handled from here? It seems like it is

**Comment by dnkoutso** [comment]
@amorde can you rebase, we fixed the integration specs problem. Rebase the integration specs PR as well.

File: lib/cocoapods/installer/analyzer.rb (hunk 2 above)
Comment (amorde) [reply lib/cocoapods/installer/analyzer.rb]: my thought was if the targets are using different user projects then they _could_ be different. super edge case, but yeah I picked the min version in that scenario

**Comment by amorde** [comment]
Build is green 🎉

**Pull request merged**